tools/dds/dds-server/dds-server.h

```diff
@@ -75,32 +83,33 @@ namespace tools
         bool prepare_devices_changed_lists(
             const rs2::event_information& info,
             std::vector< std::string >& devices_to_remove,
-            std::vector< std::pair< librealsense::dds::topics::device_info, rs2::device > >& devices_to_add );
+            std::vector< std::pair< std::string, rs2::device > >& devices_to_add );
 
-        void post_device_changes( const std::vector< std::string >& devices_to_remove,
-            const std::vector< std::pair< librealsense::dds::topics::device_info, rs2::device > >& devices_to_add );
+        void handle_device_changes( const std::vector< std::string >& devices_to_remove,
+            const std::vector< std::pair< std::string, rs2::device > >& devices_to_add );
 
 
         void remove_dds_device( const std::string& device_key );
-        void add_dds_device( const librealsense::dds::topics::device_info& dev_info, const rs2::device& rs2_dev );
-        bool verify_client_exist( const std::string& device_key,
-            std::chrono::steady_clock::duration timeout
-            = std::chrono::steady_clock::duration::zero() ) const;
+        bool add_dds_device( const std::string & device_key, const rs2::device& rs2_dev );
         bool create_device_writer( const std::string& device_key, rs2::device rs2_device );
         bool create_dds_participant( eprosima::fastdds::dds::DomainId_t domain_id );
         bool create_dds_publisher();
-        void post_connected_devices_on_wakeup();
+        void post_current_connected_devices();
+        bool send_device_info_msg( const librealsense::dds::topics::device_info& dev_info );
 
         librealsense::dds::topics::device_info query_device_info( const rs2::device& rs2_dev ) const;
         void fill_device_msg( const librealsense::dds::topics::device_info& dev_info, librealsense::dds::topics::raw::device_info& msg ) const;
 
-        std::atomic_bool _running;
+        std::atomic_bool _running, _trigger_msg_send;
         eprosima::fastdds::dds::DomainParticipant* _participant;
         eprosima::fastdds::dds::Publisher* _publisher;
         eprosima::fastdds::dds::Topic* _topic;
         eprosima::fastdds::dds::TypeSupport _topic_type;
-        std::unordered_map< std::string, dds_device_info > _devices_writers;
+        std::unordered_map< std::string, dds_device_handle > _devices_writers;
```

Comment:
Maybe we should also rename the map? `_device_handle_by_sn`?

Reply:
Sounds good, will do.